rust/agama-lib/src/storage/client/zfcp.rs

```diff
@@ -23,45 +23,55 @@
 use std::collections::HashMap;
 
 use futures_util::future::join_all;
-use zbus::{fdo::ObjectManagerProxy, zvariant::OwnedObjectPath, Connection};
+use zbus::{
+    fdo::{IntrospectableProxy, ObjectManagerProxy},
+    zvariant::OwnedObjectPath,
+    Connection,
+};
 
 use crate::{
     dbus::{extract_id_from_path, get_property},
     error::ServiceError,
     storage::{
         model::zfcp::{ZFCPController, ZFCPDisk},
-        proxies::{ZFCPControllerProxy, ZFCPManagerProxy},
+        proxies::zfcp::{ControllerProxy, ManagerProxy},
     },
 };
 
-const ZFCP_CONTROLLER_PREFIX: &'static str = "/org/opensuse/Agama/Storage1/zfcp_controllers";
+const ZFCP_CONTROLLER_PREFIX: &str = "/org/opensuse/Agama/Storage1/zfcp_controllers";
 
 /// Client to connect to Agama's D-Bus API for zFCP management.
 #[derive(Clone)]
 pub struct ZFCPClient<'a> {
-    manager_proxy: ZFCPManagerProxy<'a>,
-    object_manager_proxy: ObjectManagerProxy<'a>,
     connection: Connection,
+    manager_proxy: ManagerProxy<'a>,
+    object_manager_proxy: ObjectManagerProxy<'a>,
+    introspectable_proxy: IntrospectableProxy<'a>,
 }
 
 impl<'a> ZFCPClient<'a> {
     pub async fn new(connection: Connection) -> Result<Self, ServiceError> {
-        let manager_proxy = ZFCPManagerProxy::new(&connection).await?;
+        let manager_proxy = ManagerProxy::new(&connection).await?;
         let object_manager_proxy = ObjectManagerProxy::builder(&connection)
             .destination("org.opensuse.Agama.Storage1")?
             .path("/org/opensuse/Agama/Storage1")?
             .build()
             .await?;
+        let introspectable_proxy = IntrospectableProxy::builder(&connection)
+            .destination("org.opensuse.Agama.Storage1")?
+            .path("/org/opensuse/Agama/Storage1")?
+            .build()
+            .await?;
         Ok(Self {
+            connection,
             manager_proxy,
             object_manager_proxy,
-            connection,
+            introspectable_proxy,
         })
     }
 
     pub async fn supported(&self) -> Result<bool, ServiceError> {
-        let introspect = self.manager_proxy.introspect().await?;
-        // simply check if introspection contain given interface
+        let introspect = self.introspectable_proxy.introspect().await?; // simply check if introspection contain given interface
```

Comment:
hmm, this move of comment is wrong. It is related to the next line, not this one.